sphinx/util/i18n.py

```diff
@@ -101,8 +107,9 @@ def find_catalog_files(docname, srcdir, locale_dirs, lang, compaction):
 
 
 def find_catalog_source_files(locale_dirs, locale, domains=None, gettext_compact=False,
-                              charset='utf-8', force_all=False):
-    # type: (List[unicode], unicode, List[unicode], bool, unicode, bool) -> Set[CatalogInfo]
+                              charset='utf-8', force_all=False,
+                              excluded=lambda path: False):
+    # type: (List[unicode], unicode, List[unicode], bool, unicode, bool, Union[Callable[[unicode], bool], Matcher]) -> Set[CatalogInfo]  # NOQA
```

Comment:
How about using `Matcher([])`? I think it makes type annotation simple.

Reply:
True.